packages/kernel-5.10/kernel-5.10.spec

```diff
@@ -109,10 +109,18 @@ rpmkeys --checksig %{S:0} --dbpath "${PWD}/rpmdb"
 rm -rf "${PWD}/rpmdb"
 rpm2cpio %{S:0} | cpio -iu linux-%{version}.tar config-%{_cross_arch} "*.patch" kernel.spec
 tar -xof linux-%{version}.tar; rm linux-%{version}.tar
+# Count all the patches extracted from the SRPM
+patches_count=$(find -name "*.patch" | wc -l)
 # Find patch ordering based on the Source0 kernel.spec file from the SRPM.
 # First, find all `PatchNNN` lines. Then, sort by the patch number (-k1.6 in sort sets the 6th char
 # in field 1 of input as the sort parameter). Finally, capture just the patch file name specified.
 readarray -t patches < <(grep -P "^Patch\d+" kernel.spec | sort -n -k1.6 | grep -oP "^Patch\d+: \K.*\.patch$" kernel.spec)
+# Fail the build if there was at least one patch file found but none were found in the kernel.spec
+if [[ "${patches_count}" -gt 0 && "${#patches[@]}" == 0 ]]; then
+  echo "Mismatch on patches count!"
+  exit 1
+fi
```

Comment:
Let's assert more strongly that the number of patches in the array equals the number we extracted.